ArduCopter/takeoff.cpp

```diff
@@ -35,8 +35,9 @@ bool Mode::do_user_takeoff(float takeoff_alt_cm, bool must_navigate)
         return false;
     }
 
-    // Helicopters should return false if MAVlink takeoff command is received while the rotor is not spinning
-    if (motors->get_spool_state() != AP_Motors::SpoolState::THROTTLE_UNLIMITED && copter.ap.using_interlock) {
+    // Vehicles using motor interlock should return false if motor interlock is disabled.
+    // Interlock must be enabled to allow the controller to spool up the motor(s) for takeoff.
+    if (!motors->get_interlock() && copter.ap.using_interlock) {
```

Comment:
As a side note we've got this mix in the code where we sometimes check:

> motors->get_interlock() && copter.ap.using_interlock

.. while other times we just check 

> motors->get_interlock()

Functionally I think we shouldn't need to check the ap.using_interlock and the mix means that we will get different parts of the system reacting differently if the user changes RCx_OPTION value (on multicopters) enabling/disabling interlock.

Reply:
I've created an issue here to capture the problem. https://github.com/ArduPilot/ardupilot/issues/18408